src/CaptureClient/CaptureEventProcessorTest.cpp

```diff
@@ -237,16 +207,19 @@ static void CanHandleOneCallstackSampleOfType(Callstack::CallstackType type) {
   callstack_sample->set_timestamp_ns(100);
 
   uint64_t actual_callstack_id = 0;
-  CallstackInfo actual_callstack;
+  std::optional<CallstackInfo> actual_callstack;
   EXPECT_CALL(listener, OnUniqueCallstack)
       .Times(1)
-      .WillOnce(DoAll(SaveArg<0>(&actual_callstack_id), SaveArg<1>(&actual_callstack)));
+      .WillOnce([&](uint64_t id, CallstackInfo callstack) {
+        actual_callstack_id = id;
+        actual_callstack = std::move(callstack);
+      });
   std::optional<CallstackEvent> actual_callstack_event;
   EXPECT_CALL(listener, OnCallstackEvent).Times(1).WillOnce(SaveArg<0>(&actual_callstack_event));
 
   event_processor->ProcessEvent(event);
 
-  ExpectCallstackSamplesEqual(*actual_callstack_event, actual_callstack_id, actual_callstack,
+  ExpectCallstackSamplesEqual(*actual_callstack_event, actual_callstack_id, *actual_callstack,
```

Comment:
Does this cause a crash if `OnUniqueCallstack` is not called? In other words, do we need some `ASSERT_`?